File: september30_bolean_matrix.py
#User function Template for python3


#Function to modify the matrix such that if a matrix cell matrix[i][j]
#is 1 then all the cells in its ith row and jth column will become 1.

class Solution:
    def makeOne(self, matrix, k, d, n, m):
        for i in range(m):
            matrix[k][i] = 1
        for i in range(n):
            matrix[i][d] = 1

def booleanMatrix(matrix):
    ans = []
    n = len(matrix)
    m = len(matrix[0])
    for i in range(n):
        for j in range(m):
            if matrix[i][j] == 1:
                ans.append((i, j))

    for p in ans:
        Solution().makeOne(matrix, p[0], p[1], n, m)

#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    t = int(input())
    for _ in range(t):
        r,c = map(int, input().strip().split())
        matrix = []
        for i in range(r):
            line = [int(x) for x in input().strip().split()]
            matrix.append(line)
        booleanMatrix(matrix)
        for i in range(r):
            for j in range(c):
                print(matrix[i][j], end=' ')
            print()


# } Driver Code Ends

File: september30_bolean_matrix.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends

class Solution
{
public:
    void makeOne(vector<vector<int>> &matrix, int k, int d, int n, int m)
    {
        for (int i = 0; i < m; i++)
        {
            matrix[k][i] = 1;
        }
        for (int i = 0; i < n; i++)
        {
            matrix[i][d] = 1;
        }
    }

    // Function to modify the matrix such that if a matrix cell matrix[i][j]
    // is 1 then all the cells in its ith row and jth column will become 1.
    void booleanMatrix(vector<vector<int>> &matrix)
    {
        stack<pair<int, int>> ans;
        int n = matrix.size(), m = matrix[0].size();
        for (int i = 0; i < n; i++)
        {
            for (int j = 0; j < m; j++)
            { // Fix the loop here
                if (matrix[i][j] == 1)
                    ans.push({i, j});
            }
        }
        while (!ans.empty())
        {
            pair<int, int> p = ans.top();
            ans.pop();
            makeOne(matrix, p.first, p.second, n, m);
        }
    }
};

//{ Driver Code Starts.
int main()
{
    int t;
    cin >> t;
    while (t--)
    {
        int row, col;
        cin >> row >> col;
        vector<vector<int>> matrix(row);
        for (int i = 0; i < row; i++)
        {
            matrix[i].assign(col, 0);
            for (int j = 0; j < col; j++)
            {
                cin >> matrix[i][j];
            }
        }

        Solution ob;
        ob.booleanMatrix(matrix);

        for (int i = 0; i < row; ++i)
        {
            for (int j = 0; j < col; ++j)
            {
                cout << matrix[i][j] << " ";
            }
            cout << endl;
        }
    }
    return 0;
}

// } Driver Code Ends